src/Persistence/Sql.php

```diff
@@ -369,6 +369,12 @@ protected function setLimitOrder(Model $model, Query $query)
     public function initQueryConditions(Model $model, Query $query): void
     {
         $this->_initQueryConditions($query, $model->scope());
+
+        // add entity ID to scope to allow easy traversal
+        if ($model->id_field && $model->getId() !== null) {
+            $query->group($model->getField($model->id_field));
+            $query->having($model->getField($model->id_field), $model->getId());
```

Comment:
FYI: `having` with `group by` is like `where`, but it does not apply to aggregate functions.